### ISSUE TITLE
Gem works differently depending on source (rubygems,git)

### ISSUE THREAD

**Issue by niedfelj** [bundler/bundler#5666]
The better_errors gem on rails 5.1.1, ruby 2.4.1, bundler 1.14.6 appears to behave differently depending on if it is loaded using 

```
gem 'better_errors'
```
vs 
```
gem 'better_errors', git: 'https://github.com/charliesome/better_errors.git', branch: 'master'
```
Even if they are EXACTLY the same. 

More details here

https://github.com/charliesome/better_errors/issues/371

Wondering if you guys have any insight into why?

**Comment by segiddins** [comment]
Can you please share a project we can use to reproduce the discrepancy? Thanks!

**Comment by niedfelj** [comment]
@segiddins Sure! 

https://github.com/niedfelj/rails_51_better_errors

Try running that, changing from the git version of better_errors and the rubygems version. You can hit localhost:3000/test to throw an error. It should show the template location and proper stack trace if you are using the git version.

**Comment by segiddins** [comment]
If I point using `git` to the `v2.1.1` tag, the issue with the released gem shows up, so my guess is this is just a bug that they've fixed on `master` since the release of `v2.1.1`

**Comment by niedfelj** [comment]
Interesting. When I diffed the gem locally with my checkout, they appeared the same, but you're right that the diff does show changes here

https://github.com/charliesome/better_errors/compare/v2.1.1...master
